features/admin.tenants.v1/components/edit-self-organization/edit-self-organization-form.tsx

```diff
@@ -0,0 +1,276 @@
+    ...rest
+}: EditSelfOrganizationFormProps): ReactElement => {
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+
+    const { data: organization, mutate: mutateOrganization } = useGetSelfOrganization();
+
+    const formatDate = (dateString: string): string => {
+        if (!dateString) return "";
+
+        return moment(dateString).format("YYYY-MM-DD HH:mm:ss");
+    };
+
+    /**
+     * Handles the self organization update form submit action.
+     * @param values - Form values.
+     */
+    const handleSubmit = (values: EditSelfOrganizationFormValues): void => {
+        const operations: Array<{
+            operation: "REPLACE";
+            path: string;
+            value: string;
+        }> = [
+            {
+                operation: "REPLACE" as const,
+                path: "/name",
+                value: values.name
+            }
+        ];
+
+        updateSelfOrganization(operations)
+            .then(() => {
+                dispatch(
+                    addAlert({
+                        description: t("tenants:editSelfOrganization.notifications" +
+                            ".updateOrganization.success.description"),
+                        level: AlertLevels.SUCCESS,
+                        message: t("tenants:editSelfOrganization.notifications" +
+                            ".updateOrganization.success.message")
+                    })
+                );
+                mutateOrganization();
+                onSubmit && onSubmit();
+            })
+            .catch(() => {
+                dispatch(
+                    addAlert({
+                        description: t("tenants:editSelfOrganization.notifications" +
+                            ".updateOrganization.error.description"),
+                        level: AlertLevels.ERROR,
+                        message: t("tenants:editSelfOrganization.notifications" +
+                            ".updateOrganization.error.message")
+                    })
+                );
+            });
+    };
+
+    /**
+     * Handles the form level validation.
+     * @param values - Form values.
+     * @returns Form errors.
+     */
+    const handleValidate = (values: EditSelfOrganizationFormValues): EditSelfOrganizationFormErrors => {
+        const errors: EditSelfOrganizationFormErrors = {
+            name: undefined
+        };
+
+        if (!values.name) {
+            errors.name = t("tenants:editSelfOrganization.form.fields.organizationName.validations.required");
+        }
+
+        return errors;
+    };
+
+    if (!organization) {
+        return <div>Loading...</div>;
+    }
+
+    return (
+        <FinalForm
+            initialValues={ {
+                created: formatDate(organization.created),
+                id: organization.id,
+                lastModified: formatDate(organization.lastModified),
+                name: organization.name,
+                orgHandle: organization.orgHandle || ""
```

Comment:
Let's use optional chaining for id and name property access, otherwise this may result in something went wrong error intermittently.